lib/warnings.js

```diff
@@ -87,6 +87,11 @@ const FSTDEP021 = createDeprecation({
   message: 'The `reply.redirect()` method has a new signature: `reply.redirect(url: string, code?: number)`. It will be enforced in `fastify@v5`'
 })
 
+const FSTDEP022 = createDeprecation({
+  code: 'FSTDEP021',
+  message: 'You are using the deprecated json shorthand schema. Specify full object schema instead. It will be removed in `fastify@v5`'
+})
```

Comment:
I know, this would require some work to propagate the route's options but I think it would be helpful - just a suggestion to be more dev friendly

```suggestion
const FSTDEP022 = createDeprecation({
  code: 'FSTDEP021',
  message: 'You are using the deprecated json shorthand schema on route %s. Specify full object schema instead. It will be removed in `fastify@v5`',
  unlimited: true
})
```